### PR TITLE
set up restart and restart+clear menu items

### PR DESCRIPTION
**Pull request by rgbkrk**
Sets up restart and the restart+clear menu items for the web app.

### CODE REVIEW

File: packages/core/src/components/notebook-menu/index.js [hunk 1]
```diff
@@ -285,6 +300,18 @@ class PureNotebookMenu extends React.Component<Props> {
...
+              Restart
+            </MenuItem>
+            <MenuItem
+              key={createActionKey(MENU_ITEM_ACTIONS.RESTART_AND_CLEAR_OUTPUTS)}
+            >
+              Restart and Clear All Cells
+            </MenuItem>
+            {/*
+              <Divider />
+              <SwitchToKernelSpecsMenu />
```
Comment (rgbkrk) [inline packages/core/src/components/notebook-menu/index.js:313]: Placeholder for #2483

File: packages/core/src/epics/kernel-lifecycle.js [hunk 2]
```diff
@@ -126,7 +126,22 @@ export const restartKernelEpic = (action$: ActionsObservable<*>, store: *) =>
     concatMap(action => {
       const state = store.getState();
       const kernel = selectors.currentKernel(state);
-      const notificationSystem = state.app.notificationSystem;
+      const notificationSystem = state.app.notificationSystem
+        ? state.app.notificationSystem
+        : {
```
Comment (rgbkrk) [inline packages/core/src/epics/kernel-lifecycle.js]: The web app isn't invoking / creating the notification system, so I set up a hokey one here. I could make this part of the default state tree so we at least have it defined (though we still need to move to our own managed notification system).

Comment (rgbkrk) [reply packages/core/src/epics/kernel-lifecycle.js]: Ok, I've done that just now. Just a heads up here @theengineear.

Comment (theengineear) [reply packages/core/src/epics/kernel-lifecycle.js]: heh, nice. good enough for now!

File: packages/types/src/core/records.js [hunk 3]
```diff
@@ -167,7 +167,20 @@ export const makeAppRecord: RecordFactory<AppRecordProps> = Record({
   kernel: null,
   host: null,
   githubToken: null,
-  notificationSystem: null,
+  notificationSystem: {
+    addNotification: msg => {
```
Comment (theengineear) [inline packages/types/src/core/records.js:171]: heh, nice. good enough for now!

**Pull request merged**